ts/features/pn/reminderBanner/components/__test__/PnBannerFlowComponents.test.tsx

```diff
@@ -0,0 +1,226 @@
+import { RenderAPI, fireEvent } from "@testing-library/react-native";
+import * as React from "react";
+import { createStore } from "redux";
+import { IOScrollViewActions } from "../../../../../components/ui/IOScrollView";
+import { useIONavigation } from "../../../../../navigation/params/AppParamsList";
+import ROUTES from "../../../../../navigation/routes";
+import { applicationChangeState } from "../../../../../store/actions/application";
+import * as USEIO from "../../../../../store/hooks";
+import { appReducer } from "../../../../../store/reducers";
+import * as REMOTE_CONFIG from "../../../../../store/reducers/backendStatus/remoteConfig";
+import { renderScreenWithNavigationStoreContext } from "../../../../../utils/testWrapper";
+import { openWebUrl } from "../../../../../utils/url";
+import { sendBannerMixpanelEvents } from "../../../analytics/activationReminderBanner";
+import PN_ROUTES from "../../../navigation/routes";
+import { dismissPnActivationReminderBanner } from "../../../store/actions";
+import { PnBannerFlowComponents } from "../PnBannerFlowComponents";
+
+const { LoadingScreen, SuccessScreen, ErrorScreen, CtaScreen } =
+  PnBannerFlowComponents;
+
+jest.mock("../../../../../utils/hooks/useOnFirstRender", () => ({
+  useOnFirstRender: (callback: () => void) => {
+    callback();
+  }
+}));
+
+jest.mock("../../../../../utils/url", () => ({
+  openWebUrl: jest.fn()
+}));
+
+jest.mock("../../../analytics/activationReminderBanner", () => ({
+  sendBannerMixpanelEvents: {
+    alreadyActive: jest.fn(),
+    activationSuccess: jest.fn(),
+    bannerKO: jest.fn()
+  }
+}));
+
+jest.mock("../../../../../navigation/params/AppParamsList", () => ({
+  useIONavigation: jest.fn().mockReturnValue({
+    navigate: jest.fn()
+  })
+}));
+
+const mockPrivacyUrl = "https://example.com/privacy";
+const mockTosUrl = "https://example.com/tos";
+
+describe("PnBannerFlowComponents", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("LoadingScreen", () => {
+    it("should display loading component with correct title for LOADING-ACTIVATION state", () => {
+      const flowState = "LOADING-ACTIVATION";
+      const { toJSON, getByTestId } = renderLoadingScreen(flowState);
+      expect(getByTestId(`loading-${flowState}`)).toBeTruthy();
+      expect(toJSON()).toMatchSnapshot();
+    });
+
+    it("should display loading component with correct title for LOADING-DATA state", () => {
+      const flowState = "LOADING-DATA";
+      const { toJSON, getByTestId } = renderLoadingScreen(flowState);
+      expect(getByTestId(`loading-${flowState}`)).toBeTruthy();
+      expect(toJSON()).toMatchSnapshot();
+    });
+  });
+
+  describe("SuccessScreen", () => {
+    afterEach(() => {
```

Comment:
Theoretically, we can remove the `beforeEach` on line 49 and replace it with this `afterEach` (moving it to line 49 so it is global to the suite)